ddl/placement_policy.go

```diff
@@ -65,15 +67,62 @@ func onCreatePlacementPolicy(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64
 	}
 }
 
-func checkPolicyValidation(info *placementpolicy.PolicyInfo) error {
+func checkPolicyLabels(w *worker, rules []*placement.Rule) error {
+	distinctLabels := make(map[string]struct{})
+	for _, rule := range rules {
+		if rule.Constraints != nil {
+			for _, cons := range rule.Constraints {
+				distinctLabels[cons.Key] = struct{}{}
+			}
+		}
+	}
+	if len(distinctLabels) == 0 {
+		return nil
+	}
+	sysSe, err := w.sessPool.get()
+	if err != nil {
+		return err
+	}
+	defer w.sessPool.put(sysSe)
+	stmt, err := sysSe.(sqlexec.RestrictedSQLExecutor).ParseWithParams(context.Background(), "show placement labels")
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	rows, _, err := sysSe.(sqlexec.RestrictedSQLExecutor).ExecRestrictedStmt(context.Background(), stmt)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	b := &placementpolicy.CheckPolicyLabelsResultBuilder{}
+	for _, row := range rows {
+		key := row.GetString(0)
+		valueList := row.GetJSON(1)
+		if err := b.AppendShowLabels(key, valueList); err != nil {
+			return errors.Trace(err)
+		}
+	}
+	err = b.CheckLabels(distinctLabels)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func (w *worker) checkPolicyConstraints(info *placementpolicy.PolicyInfo) error {
+	var totalRules []*placement.Rule
```

Comment:
Many logics are duplicated with pr: #27814 . Can we reuse the code after one committed?